clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslServerAuthenticator.java

```diff
@@ -73,6 +73,7 @@
 
 public class SaslServerAuthenticator implements Authenticator {
 
+    private static final int MAX_RECEIVE_SIZE = 131072; // Chosen as twice the max receive size for GSSAPI
```

Comment:
If we are going to limit receive size to a fixed value, we could probably add the limit to `SaslClientAuthenticator` as well?